### PR TITLE
Added Sourcemaps, Changed Task Naming Convention, Used return With gulp....

### PR DESCRIPTION
**Pull request by bensternthal**
...src()

@craigcook 

This should work as intended. Checkout then:
1. npm install
2. gulp

Sourcemaps should be generated.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -1,36 +1,45 @@
...
-gulp.task('compress-base', function () {
-  var filter = gulpFilter(['*', '!oldIE.styl']);
+var paths = {
+  stylus: './masterfirefoxos/base/static/stylus',
+  css: './masterfirefoxos/base/static/css/'
+};
 
-  gulp.src('./masterfirefoxos/base/static/stylus/**.styl')
-    .pipe(filter)
+gulp.task('compress::base', function () {
```
Comment (jgmize) [inline gulpfile.js]: Why the change from `-` to `::`?

File: gulpfile.js [hunk 2]
```diff
@@ -1,36 +1,45 @@
...
-gulp.task('watch', ['compress'], function () {
-  gulp.watch("./masterfirefoxos/base/static/stylus/**.styl", ['compress']);
+gulp.task('clean:css', function (cb) {
+  return del([
+    paths.css + '/*',
+    '!' + paths.css + '/README.md'
+  ], cb);
 });
 
+gulp.task('default', ['clean:css', 'compress::base', 'compress::ie']);
```
Comment (jgmize) [inline gulpfile.js]: In your new naming convention, what are the criteria for a single vs a double colon?

Comment (bensternthal) [inline gulpfile.js]: This is me not being careful. Should only be one colon.

I saw 2 conventions used in the gulp "recipes" examples:

1. foo-bar
2. foo:bar

Personally I like 2, the colon to me is just clearer that there could be more than one target for the task... example clean:css clean:js rather than clean-css, clean-js

I dont feel strongly about this.. so if you want the dash we can go that way.

Comment (jgmize) [inline gulpfile.js]: Ah, that makes more sense now. The double colon was throwing me off, but I don't have any strong opinions on a single colon vs dash.

Comment (bensternthal) [inline gulpfile.js]: After a spirited discussion in IRC with the delightful alex g, we have decided to go with a single colon.

I will adjust my pr (and rebase with the latest code).

**Comment by bensternthal** [comment]
Closing so i can commit a cleaner PR.
